### PR TITLE
Fix commit Ids for webhooks to ArgoCd

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**

### CODE REVIEW

File: services/cd-service/pkg/repository/repository.go [hunk 1]
```diff
@@ -750,19 +743,31 @@ func (r *repository) ApplyTransformers(ctx context.Context, transformers ...Tran
 	var rev *git.Oid
 	if state.Commit != nil {
 		rev = state.Commit.Id()
+	} else {
+		// first commit in the repo!
```
Comment (leandro-freiheit) [inline services/cd-service/pkg/repository/repository.go:747]: The comment is relevant but I don't like having an empty else statement

**Review by leandro-freiheit — APPROVED** [review]

File: services/cd-service/pkg/repository/repository.go (hunk 1 above)
Comment (hannesg) [reply services/cd-service/pkg/repository/repository.go:747]: +1

File: services/cd-service/pkg/repository/repository.go [hunk 2]
```diff
@@ -676,6 +660,12 @@ type RootApp struct {
 type TransformerResult struct {
 	ChangedApps     []AppEnv
 	DeletedRootApps []RootApp
+	Commits         *Commits
+}
+
+type Commits struct {
+	Previous string
```
Comment (hannesg) [inline services/cd-service/pkg/repository/repository.go:667]: I would find it more obvious if we would stick with *git.Oid here. Is there a specific reasons why you use string? If yes, maybe document it.

**Review by hannesg — APPROVED** [review]
How about some tests?

File: services/cd-service/pkg/repository/repository.go (hunk 2 above)
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/repository.go]: done

File: services/cd-service/pkg/repository/repository.go (hunk 1 above)
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/repository.go]: done

**Pull request merged**